dashboard/app/models/script.rb

```diff
@@ -451,7 +451,20 @@ def self.add_script(options, data)
     end
     script_stages.each do |stage|
       stage.script_levels = script_levels_by_stage[stage.id]
+
+      # Go through all the script levels for this stage, except the last one,
+      # and raise an exception if any of them are a multi-page assessment.
+      # (That's when the script level is marked assessment, and the level itself
+      # has a pages property and more than one page in that array.)
+      # This is because only the final level in a stage can be a multi-page
+      # assessment.
+      stage.script_levels.each do |script_level|
```

Comment:
you could do `stage.script_levels[0..-2].any?(&:long_assessment?)` which would save you n-1 calls to `end_of_stage?`. But it's arguably more readable your way, especially if people don't remember the difference between .. and ...)